psi4/driver/qcdb/libmintsbasisset.py

```diff
@@ -520,76 +520,86 @@ def build(molecule, shells):
-#TRIAL#            mol.set_shell_by_number(at, oneatombasishash, role="CABS")
-#TRIAL#        mol.update_geometry()  # re-evaluate symmetry taking basissets into account
-#TRIAL#
-#TRIAL#        text = """   => Creating Basis Set <=\n\n"""
-#TRIAL#        text += """    Role: %s\n""" % (fitroles)
-#TRIAL#        text += """    Keyword: %s\n""" % (keys)
-#TRIAL#        text += """    Name: %s\n""" % (name)
-#TRIAL#
-#TRIAL#        if returnBasisSet:
-#TRIAL#            print(text)
-#TRIAL#            return basisset
-#TRIAL#        else:
-#TRIAL#            bsdict = {}
-#TRIAL#            bsdict['message'] = text
-#TRIAL#            bsdict['name'] = basisset.name
-#TRIAL#            bsdict['puream'] = int(basisset.has_puream())
-#TRIAL#            bsdict['shell_map'] = basisset.export_for_libmints("CABS")
-#TRIAL#            return bsdict
+    @staticmethod
+    def pyconstruct_combined(mol, keys, targets, fitroles, others):
+
+        # make sure the lengths are all the same
+        if len(keys) != len(targets) or len(keys) != len(fitroles):
+            raise ValidationError("""Lengths of keys, targets, and fitroles must be equal""")
+
+        # Create (if necessary) and update qcdb.Molecule
+        if isinstance(mol, str):
+            mol = Molecule(mol)
+            returnBasisSet = False
+        elif isinstance(mol, Molecule):
+            returnBasisSet = True
+        else:
+            raise ValidationError("""Argument mol must be psi4string or qcdb.Molecule""")
+        mol.update_geometry()
+
+        # load in the basis sets
+        sets = []
+        name = ""
+        keywords = ""
+        blends = ""
+        for at in range(len(keys)):
+            bas = BasisSet.pyconstruct(mol, keys[at], targets[at], fitroles[at], others[at])
+            name += targets[at] + " + "
+            keywords += keys[at] + " + "
+            blends += bas.name.upper() + " + "
+            sets.append(bas)
+
+        name = name[:-3].strip()
+        keywords = keywords[:-3].strip()
+        blends = blends[:-3].strip()
+
+        # work our way through the sets merging them
+        combined_atom_basis_shell = collections.OrderedDict()
+        for at in range(len(sets)):
+            atom_basis_shell = sets[at].atom_basis_shell
+
+            for label, basis_map in atom_basis_shell.items():
+                if label not in combined_atom_basis_shell:
+                    combined_atom_basis_shell[label] = collections.OrderedDict()
+                    combined_atom_basis_shell[label][name] = []
+                for basis, shells in basis_map.items():
+                    combined_atom_basis_shell[label][name].extend(shells)
+
+        # sort the shells by angular momentum
+        for label, basis_map in combined_atom_basis_shell.items():
+            combined_atom_basis_shell[label][name] = sorted(combined_atom_basis_shell[label][name],\
```

Comment:
```suggestion
            combined_atom_basis_shell[label][name] = sorted(combined_atom_basis_shell[label][name],
```
as long as there's an open parenthesis or list or similar, python doesn't need the line continuation